### PR TITLE
feature/10_token_generator

### PR DESCRIPTION
**Pull request by frbattid**
* Implements issue #10 
* No unit tests yet implemented.
* (unofficial) e2e tests passed:
```
$ /Users/frb/.nvm/v0.10.38/bin/npm start

> cosmos-auth@0.0.0 start /Users/frb/devel/fiware/fiware-cosmos/cosmos-auth
> node ./src/server.js

cosmos-auth running at http://localhost:13000
[user --> cosmos-auth] Request: GET /tidoop/v1/version
[user <-- cosmos-auth] Response: {version: 0.0.0}
[user --> cosmos-auth] Request: POST /cosmos-auth/v1/token
[user --> cosmos-auth] {"user-agent":"curl/7.37.1","host":"localhost:13000","accept":"*/*","content-type":"application/x-www-form-urlencoded","content-length":"57"}
[user --> cosmos-auth] grant_type=password&username=frb@tid.es&password=xxxxxxxx
[cosmos-auth -> IdM] Request: POST https://account.lab.fiware.org:443/oauth2/token
[cosmos-auth -> IdM] {"Authorization":"Basic xxxxxxxxxxxxxxxx","Content-Type":"application/x-www-form-urlencoded","Content-Length":57}
[cosmos-auth -> IdM] grant_type=password&username=frb@tid.es&password=xxxxxxxx
[cosmos-auth <- IdM] Response: {"access_token": "xxxxxxxx", "token_type": "Bearer", "expires_in": 3600, "refresh_token": "xxxxxxxx"}
[user <-- cosmos-auth] Response: {"access_token": "xxxxxxxx", "token_type": "Bearer", "expires_in": 3600, "refresh_token": "xxxxxxxx"}
...
$ curl -X GET "http://localhost:13000/cosmos-auth/v1/version"
{version: 0.0.0}
$ curl -X POST "http://localhost:13000/cosmos-auth/v1/token" -H "Content-Type: application/x-www-form-urlencoded" -d "grant_type=password&username=frb@tid.es&password=xxxxxxxx"
{"access_token": "xxxxxxxx", "token_type": "Bearer", "expires_in": 3600, "refresh_token": "xxxxxxxx"}
```
* Assignee @gtorodelvalle

### CODE REVIEW

File: cosmos-auth/src/client.js [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+
+    request.on('error', function(error) {
+        callback(error);
+    });
+} // doRequest
+
+
+module.exports = {
+    doAuthorizedRequest: doAuthorizedRequest
+}
```
Comment (gtorodelvalle) [inline cosmos-auth/src/client.js]: Final blank line requested... :)

File: cosmos-auth/src/server.js [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+var boom = require('boom');
+var packageJson = require('../package.json');
+var config = require('../conf/cosmos-auth.json');
+var client = require('./client.js');
+
+// Create a Hapi server with a host and port
+var server = new Hapi.Server();
+
+server.connection({
+    host: 'localhost',
```
Comment (gtorodelvalle) [inline cosmos-auth/src/server.js]: I guess you plan to make this host configurable in the future, right? Leaving it as localhost won't make it visible "from the outside world", right? :)

**Comment by gtorodelvalle** [comment]
Apart from the included comments, it looks fine to me... :)

File: cosmos-auth/src/server.js (hunk 2 above)
Comment (frbattid) [inline cosmos-auth/src/server.js]: Nice catch :)

File: cosmos-auth/src/client.js (hunk 1 above)
Comment (frbattid) [inline cosmos-auth/src/client.js]: Fixed in https://github.com/telefonicaid/fiware-cosmos/commit/859f2cfdbe9fdbc6628a6623d8555d895aacdaf8

File: cosmos-auth/src/server.js (hunk 2 above)
Comment (frbattid) [inline cosmos-auth/src/server.js]: Fixed in https://github.com/telefonicaid/fiware-cosmos/commit/859f2cfdbe9fdbc6628a6623d8555d895aacdaf8

**Comment by frbattid** [comment]
Ready for another round, thanks! :)

**Comment by gtorodelvalle** [comment]
LGTM and landing!!! Thanks @frbattid :)

**Pull request merged**